src/openeo_pg_parser/definitions.py

```diff
@@ -0,0 +1,201 @@
+import os
+from json import load
+from openeo_pg_parser.utils import load_processes
+
+
+class OpenEOProcess:
+    """ Class representing an OpenEO process definition. """
+    def __init__(self, process_def):
+        """
+        Constructor of `OpenEOProcess` class.
+
+        Parameters
+        ----------
+        process_def : dict
+            A dictionary defining an openEO process definition/schema.
+
+        """
+        self.definition = process_def
+        self.sub_process = None
+
+    @classmethod
+    def from_file(cls, filepath):
+        """
+        Creates an `OpenEOProcess` instance from a process definition (JSON) file.
+
+        Parameters
+        ----------
+        filepath : str
+            Full filepath to the process definition (JSON) file.
+
+        Returns
+        -------
+        OpenEOProcess
+
+        """
+        if os.path.splitext(filepath)[1] != 'json':
+            err_msg = "The given file must be a json file."
+            raise IOError(err_msg)
+
+        process_def = load(filepath)
+        return cls(process_def)
+
+    @classmethod
+    def from_name(cls, name, src=None):
+        """
+        Creates an `OpenEOProcess` instance from an process name/ID.
+
+        Parameters
+        ----------
+        name : str
+            The name of the process, i.e. its ID.
+        src : dict or str or list, optional
+            It can be:
+                - dictionary of loaded process definitions (keys are the process ID's)
+                - directory path to processes (.json)
+                - URL of the remote process endpoint (e.g., "https://earthengine.openeo.org/v1.0/processes")
+                - list of loaded process definitions
+
+        Returns
+        -------
+        OpenEOProcess
+
+        """
+
+        # use standard directory
+        if src is None:
+            src = os.path.join(os.path.dirname(__file__), "..", "..", "processes")
+
+        processes = load_processes(src)
+
+        if name not in processes.keys():
+            err_msg = "Process '{}' could not be found in the list of processes.".format(name)
+            raise ValueError(err_msg)
+        else:
+            return cls(processes[name])
+
+    @property
+    def id(self):
+        """ str : ID/name of the process. """
+        return self.definition['id']
+
+    @property
+    def description(self):
+        """ str : Description of the process. """
+        return self.definition['description'] if self._has_description else None
```

Comment:
All these constructions like
```python
self.definition['foo'] if self._has_foo else None
```

can be simplified in Python to a simple `.get()` call , which allows to specify a default value:
```
self.definitions.get('foo', None)
```

Moreover, the default value of this default argument is `None`, so you can write just:
```
self.definitions.get('foo')
```

Reply:
Ah yes, I know, but I completely forgot to use it in this case... Thanks a lot!